examples/billing/get_invoices.rb

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env ruby
+# Encoding: utf-8
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# This code example retrieves the invoices issued last month for a given
+# billing setup.
+
+require 'optparse'
+require 'google/ads/google_ads'
+
+def get_invoices(customer_id, billing_setup_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  # Gets the date one month before now.
+  last_month = Date.today.prev_month
+
+  # Issues a request to list invoices.
+  response = client.service.invoice.list_invoices(
+    customer_id: customer_id,
+    billing_setup: client.path.billing_setup(customer_id, billing_setup_id),
+    # The year needs to be 2019 or later.
+    issue_year: last_month.year.to_s,
+    issue_month: last_month.strftime("%B").upcase.to_sym,
+  )
+
+  # Iterates over all invoices retrieved and prints their information.
+  response.invoices.each do |invoice|
+    puts "- Found the invoice '#{invoice.resource_name}'" \
+      " ID (also known as Invoice Number): '#{invoice.id}'\n" \
```

Comment:
This giant block of text would probably be easier to read for a human and easier to edit for future maintainers if we use the heredoc format like we do for queries in other examples.

```
puts <<~OUTPUT
  ...
OUTPUT 
```

Then you could remove all the newlines, quotes, and backslashes.

Reply:
Yup good call. I've made changes to use heredoc format instead.